### PR TITLE
Insert aviation messages to database

### PR DESCRIPTION
**Pull request by kasarit**

### CODE REVIEW

**Review by petringo — CHANGES_REQUESTED** [review]
In general this looks really good! Though I would still request to configure connection timeouts and implement proper recovery on errors. Unless I just didn't see it exists.

In additions to timeouts mentioned beside the code, I remember seeing a login timeout somewhere. It probably should be set if possible. Please check if there's anything else on this matter.

File: src/main/java/fi/fmi/avi/archiver/database/DatabaseAccess.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    private final Clock clock;
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert insertAviationMessage;
+    private final SimpleJdbcInsert insertRejectedAviationMessage;
+
+    public DatabaseAccess(final NamedParameterJdbcTemplate jdbcTemplate, final Clock clock) {
+        this.clock = requireNonNull(clock, "clock");
+        this.jdbcTemplate = requireNonNull(jdbcTemplate, "jdbcTemplate");
+        this.insertAviationMessage = new SimpleJdbcInsert(jdbcTemplate.getJdbcTemplate())
```
[the remark this answers is not in the document]
Comment (kasarit) [reply src/main/java/fi/fmi/avi/archiver/database/DatabaseAccess.java:34]: These timeouts are now set using the spring.jdbc.template.query-timeout property.

File: src/test-integration/java/fi/fmi/avi/archiver/database/DatabaseAccessTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+@JdbcTest
+@Sql(scripts = {"classpath:/schema-h2.sql", "classpath:/h2-data/avidb_message_types_test.sql",
+        "classpath:/h2-data/avidb_message_format_test.sql", "classpath:/h2-data/avidb_message_routes_test.sql",
+        "classpath:/h2-data/avidb_stations_test.sql"}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+@Sql(scripts = "classpath:/h2-data/avidb_cleanup_test.sql", executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+@ContextConfiguration(classes = {AviationMessageArchiver.class, TestConfig.class},//
+        loader = AnnotationConfigContextLoader.class,//
+        initializers = {ConfigDataApplicationContextInitializer.class})
+public class DatabaseAccessTest {
```
Comment (petringo) [inline src/test-integration/java/fi/fmi/avi/archiver/database/DatabaseAccessTest.java:33]: Would there exist a reasonable way to test retry on both recoverable and non-recoverable database errors?

**Review by petringo — CHANGES_REQUESTED** [review]
Thanks, great work again! Sorry for being picky and seeing potential problems everywhere.

File: src/main/java/fi/fmi/avi/archiver/config/DataSourceConfig.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Bean
+    public RetryTemplate databaseAccessRetryTemplate() {
+        final ExponentialBackOffPolicy backOffPolicy = new ExponentialBackOffPolicy();
+        backOffPolicy.setInitialInterval(retryInitialInterval.toMillis());
+        backOffPolicy.setMultiplier(retryMultiplier);
+        backOffPolicy.setMaxInterval(retryMaxInterval.toMillis());
+
+        final RetryTemplateBuilder retryTemplateBuilder = new RetryTemplateBuilder();
+        retryTemplateBuilder.infiniteRetry();
```
[the remark this answers is not in the document]
Comment (kasarit) [reply src/main/java/fi/fmi/avi/archiver/config/DataSourceConfig.java:78]: Definitely a good idea. Makes me more comfortable about using the infinite retry to begin with.

**Review by petringo — APPROVED** [review]
There it is now.

**Pull request merged**